src/inmanta/server/services/compilerservice.py

```diff
@@ -597,6 +600,14 @@ async def notify(listener: CompileStateListener) -> None:
         await asyncio.gather(*[notify(listener) for listener in self.listeners])
         await compile.update_fields(handled=True)
         LOGGER.log(const.LOG_LEVEL_TRACE, "listeners notified for %s", compile.id)
+        if compile.success is False and compile.do_export:
+            await self._notification_service.notify(
```

Comment:
Why did we add this code in the `_notify_listeners()` method? It breaks a bit the cohesion of the method in the sense that this call has nothing to do with listeners. But maybe there is a good reason for this. Is it because this method is always invoked, even when the server restarts?

Reply:
If we did it for the above-mentioned reason, we should execute `await compile.update_fields(handled=True)` after sending the notification.

Reply:
I agree that this is not the right place, but that instead it should be a listener that this slice registers to itself or the notification service register to this slice.